ui/store/action-queue/index.test.js

```diff
@@ -0,0 +1,358 @@
+import sinon from 'sinon';
+
+import {
+  dropQueue,
+  callBackgroundMethod,
+  submitRequestToBackground,
+  _setBackgroundConnection,
+} from '.';
+
+// This file tests only MV3 queue scenario
+// MV2 tests are already covered by '../actions.test.js'
+
+jest.mock('../../../shared/modules/mv3.utils', () => {
+  return {
+    isManifestV3: () => true,
+  };
+});
+
+describe('ActionQueue', () => {
+  afterEach(() => {
+    sinon.restore();
+    dropQueue(true);
+  });
+
+  describe('dropQueue', () => {
+    it('rejects all pending actions by default', async () => {
+      const background = {
+        connectionStream: {
+          readable: false,
+        },
+        backgroundFunction: sinon.stub().yields(),
+      };
+
+      _setBackgroundConnection(background);
+      const result = submitRequestToBackground('backgroundFunction');
+      dropQueue();
+
+      await expect(result).rejects.toThrow(
+        'Background operation cancelled while waiting for connection.',
+      );
+      expect(background.backgroundFunction.called).toStrictEqual(false);
+    });
+  });
+  describe('submitRequestToBackground', () => {
+    it('calls promisified background method if the stream is connected', async () => {
+      const background = {
+        connectionStream: {
+          readable: true,
+        },
+        backgroundFunction1: sinon.stub().yields(),
+      };
+
+      _setBackgroundConnection(background);
+      await submitRequestToBackground('backgroundFunction1');
+      expect(background.backgroundFunction1.called).toStrictEqual(true);
+    });
+
+    it('does not calls promisified background method if the stream is not connected', async () => {
+      const background = {
+        connectionStream: {
+          readable: false,
+        },
+        backgroundFunction2: sinon.stub().yields(),
+      };
+
+      _setBackgroundConnection(background);
+      submitRequestToBackground('backgroundFunction2');
```

Comment:
await is not required here

Reply:
why? I dare to disagree

Reply:
This is scenario when stream is not connected, thus method is not called and is not resolved, request timeouts here if I write await.

This other test case is succeeding without await https://github.com/MetaMask/metamask-extension/pull/15337/files#diff-026267adeb51c0d42e0900bfc1f12e32bbb14e42fea7d7727dabdee5c5f4df3dR54

Reply:
Oh, it's the test checking the immediate results, not the one where connection comes back. How sis the await get here at all then?

Reply:
This test never gets into a connected state so there's no way the await would not timeout. That's fine. 

Be careful with toggling async behavior on and off in these tests, because most of them depend on that a lot and may no longer assert correct behavior.